test/js-api/memory/constructor.any.js

```diff
@@ -8,15 +8,18 @@ function assert_Memory(memory, expected) {
 
   // https://github.com/WebAssembly/spec/issues/840
   assert_equals(memory.buffer, memory.buffer, "buffer should be idempotent");
-  assert_equals(Object.getPrototypeOf(memory.buffer), ArrayBuffer.prototype,
-                "prototype of buffer");
-  assert_true(Object.isExtensible(memory.buffer), "buffer extensibility");
+  const isShared = !!expected.shared;
+  const bufferType = isShared ? SharedArrayBuffer : ArrayBuffer;
```

Comment:
I think this is going to cause a `ReferenceError` if `SharedArrayBuffer` is not present, even in the not-shared case. Can you use `self.SharedArrayBuffer` instead?